relay-system/src/service.rs

```diff
@@ -1025,6 +1022,45 @@ pub trait Service: Sized {
     }
 }
 
+/// Keeps track of running services.
+///
+/// Exposes information about crashed services.
+#[derive(Debug, Default)]
+pub struct ServiceRunner(FuturesUnordered<JoinHandle<()>>);
```

Comment:
Tokio has a [`JoinSet`](https://docs.rs/tokio/1.41.1/tokio/task/struct.JoinSet.html) for awaiting/spawning tasks, can use that instead of `FuturesUnordered`.

Reply:
@Dav1dde IIUC, the JoinSet requires you to call `JoinSet::spawn`, which means we would not be able to use `relay_system::spawn!`.